core/services/workflows/syncer/heap.go

```diff
@@ -0,0 +1,63 @@
+package syncer
+
+import "container/heap"
+
+type Heap interface {
```

Comment:
Future style note: it's best to abstract as late as possible and to avoid altogether if possible; in this case you have exactly one implementation of the heap (blockHeighHeap), so you could have avoided declaring these at all (I think anyway -- I'm not sure I follow why publicHeap is necessary).